### PR TITLE
usability and fixes

### PR DESCRIPTION
**Pull request by thomasmckay**
+ import/export sync plans
+ added --organization to some resources
+ export and import red hat products in addition to custom ones to allow them to be sync'ed
+ import updated to include products and subscriptions properly
+ fixed products cli args
+ updated user admin flag

### CODE REVIEW

File: lib/hammer_cli_csv/base.rb [hunk 1]
```diff
@@ -756,7 +796,7 @@ def associate_locations(id, locations, name)
     def search_string(resource, name)
       operator = case resource
                  when "gpg-key", "sync-plan", "lifecycle-environment", "host-collections"
-                   @server_status['version'] && @server_status['version'].starts_with?('1.7') ? '=' : ':'
+                   @server_status['version'] && @server_status['version'].match(/^1\.7/) ? '=' : ':'
```
Comment (komidore64) [inline lib/hammer_cli_csv/base.rb]: prefer `\A` and `\z` over `^` and `$`, respectively. so this should be `match(/\A1\.7/)`.

[reference](http://stackoverflow.com/a/577675/336520)

File: lib/hammer_cli_csv/sync_plans.rb [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+# http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+
+
+module HammerCLICsv
+  class CsvCommand
+    class SyncPlansCommand < BaseCommand
+      command_name 'sync-plans'
+      desc         'import or export repository sync plans'
+
+      option %w(--organization), 'ORGANIZATION', 'Only process organization matching this name'
```
Comment (komidore64) [inline lib/hammer_cli_csv/sync_plans.rb]: i'm seeing this all over the place. would you be able to create a base csv command that contains all the similar code?

File: lib/hammer_cli_csv/sync_plans.rb [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+            # TODO
+            # @api.resource(:host_collections).call(:update, {
+            #     'organization_id' => line[ORGANIZATION],
+            #     'id' => @existing[line[ORGANIZATION]][name],
+            #     'name' => name,
+            #     'max_systems' => (line[LIMIT] == 'Unlimited') ? -1 : line[LIMIT],
+            #     'description' => line[DESCRIPTION]
+            # })
+          end
+          print "done\n" if option_verbose?
```
Comment (komidore64) [inline lib/hammer_cli_csv/sync_plans.rb]: why not `puts`?

File: lib/hammer_cli_csv/sync_plans.rb (hunk 2 above)
Comment (thomasmckay) [inline lib/hammer_cli_csv/sync_plans.rb]: Not all resources accept --organization (foreman objects). I originally had it in base.rb and may move it back there when org becomes optional for katello objects.

Comment (komidore64) [inline lib/hammer_cli_csv/sync_plans.rb]: i gotcha

**Comment by thomasmckay** [comment]
@komidore64 updated

**Comment by komidore64** [comment]
cool

ack

**Pull request merged**